tests/collections/asr/test_asr_ctcencdec_model.py

```diff
@@ -114,3 +110,48 @@ def test_vocab_change(self, asr_model):
         asr_model.change_vocabulary(new_vocabulary=new_vocab)
         # fully connected + bias
         assert asr_model.num_weights == nw1 + 3 * (asr_model.decoder._feat_in + 1)
+
+    @pytest.mark.unit
+    def test_dataclass_instantiation(self, asr_model):
+        model_cfg = configs.EncDecCTCModelConfig()
+
+        # Update mandatory values
+        vocabulary = asr_model.decoder.vocabulary
+        model_cfg.model.labels = vocabulary
+
+        # Update encoder
+        model_cfg.model.encoder.activation = 'relu'
+        model_cfg.model.encoder.feat_in = 64
+        model_cfg.model.encoder.jasper = [
+            configs.JasperEncoderConfig(
+                filters=1024,
+                repeat=1,
+                kernel=[1],
+                stride=[1],
+                dilation=[1],
+                dropout=0.0,
+                residual=False,
+                se=True,
+                se_context_size=-1,
+            )
+        ]
+
+        # Update decoder
+        model_cfg.model.decoder.feat_in = 1024
+        model_cfg.model.decoder.num_classes = 28
+        model_cfg.model.decoder.vocabulary = vocabulary
+
+        # Construct the model
+        asr_cfg = OmegaConf.create({'model': asr_model.cfg})
+        model_cfg = EncDecCTCModel.update_model_dataclass(model_cfg, asr_cfg)
```

Comment:
This line is just to test the function, we can just pass asr_cfg into EncDecCTCModel and it gives same thing (but with the 3 data loaders which need to have their manifest filepath set to None explicitly + the optim/sched section).